kiwi_lint/function_based_views.py

```diff
@@ -0,0 +1,96 @@
+"""function_based_views.py provides a pylint checker FunctionBasedViewChecker that warns
+    against using function based views in Django.
+"""
+
+from importlib import import_module
+
+import django
+from django.conf import settings
+from django.urls.resolvers import URLResolver, URLPattern
+
+from pylint import checkers
+from pylint import interfaces
+
+
+class DjangoViewsVisiter(checkers.BaseChecker):
+    """Base class for visiting only astroid modules which contain django views.
+
+    Instance Attributes:
+        view_module
+        Type :: union[string|None]
+
+            view_module == None,        if the current astroid module does _not_ contain views
+            view_module == module.name, if the current astroid module contains _routed_ views
+
+        url_mapping
+        Type :: dict[url_pattern: str, 2-tuple(module_name: str, view_name: str)]
+
+        view_files
+        Type :: set[view_file_paths: string]
+    """
+
+    def __init__(self, linter):
+        super().__init__(linter)
+
+        django.setup()
+
+        project_urls = import_module(settings.ROOT_URLCONF)
+        self.url_mapping = self._get_url_view_mapping(project_urls.urlpatterns)
+        self.view_files = set(module for module, _ in self.url_mapping.values())
+        self.view_module = None
+
+    @classmethod
+    def _get_url_view_mapping(cls, root_urlpatterns):
+        def go(urlpatterns, prefix='^', result=None):
+            """Flattens the url graph
+
+                Returns a dictionary of the url pattern string as a key
+                and tuple of the module name and the view name
+            """
+
+            if result is None:
+                result = {}
+
+            for url in urlpatterns:
+                if isinstance(url, URLPattern):
+                    # path('someurl', view), meaning this is leaf node url
+                    url_pattern = prefix + url.pattern.regex.pattern.strip('^')
+                    result[url_pattern] = (url.callback.__module__, url.callback.__name__)
```

Comment:
`result[rl.callback.__module__] = url.callback.__name__` will directly provide the data structure expected by the checker and negate the need for `group_views_by_module()` below.